lab_1/main.py

```diff
@@ -98,22 +207,34 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
-
-
-def load_profile(path_to_file: str) -> dict or None:
-    """
-    Loads a language profile
-    :param path_to_file: a path
-    :return: a dictionary with three keys – name, freq, n_words
-    """
-    pass
-
-
-def save_profile(profile: dict) -> int:
-    """
-    Saves a language profile
-    :param profile: a dictionary
-    :return: 0 if everything is ok, 1 if not
-    """
-    pass
+    try:
+        if not languages:
+            languages = profiles.copy()
+        count = 0
+        score_best = 0
+        profile = profiles[count]
+        score_best_name = profile["name"]
+        score_current = 0
+        check = True
+        for language in languages:
+            if isinstance(language, dict):
+                language = language["name"]
+            if language == profile["name"]:
+                score_current = compare_profiles_advanced(unknown_profile, profile, top_n)["score"]
```

Comment:
Do we work with only one known profile?
I don't get it

Reply:
220-221: The incoming checks made me create these lines since the code would only work either for many elements in "languages" or one. This resolved the issue.

222-223: I have languages and profiles. If one element of the former (which is called language, singular) is in the latter (I compare it with one profile in profiles) at this point of code execution, then the calculation of the score happens and possible best_score update is following.